prs/WeatherData.cs

```diff
@@ -0,0 +1,35 @@
+namespace Sample {
+    /**
+     * A class which updates displays with various atmospheric data.
+     */
+    public class WeatherData {
+
+        private float temperature;
+        private float humidity;
+        private float pressure;
+        private Location location;
+
+        public WeatherData(float temperature, float humidity, float pressure, Location location) {
+            this.temperature = temperature;
+            this.humidity = humidity;
+            this.pressure = pressure;
+            this.location = location;
+        }
+
+        public void updateMeasurements(float? temperature, float? humidity, float? pressure) {
+            this.temperature = temperature ?? this.temperature;
+            this.humidity = humidity ?? this.humidity;
+            this.pressure = pressure ?? this.pressure;
+
+            updateTargets(temperature, humidity, pressure);
+        }
+
+        // Whenever there's a new target, just add it to this private method 😁
```

Comment:
Ideally, I'd love for our service to be a little more flexible than this. If we want other targets to really use our service, it could be a good use case for the observer pattern?